### PR TITLE
[BugFix] Addressing Multi-valued inflection point issues

### PR DESCRIPTION
**Pull request by bregnery**
## Description
Previously the inflection point finder allowed for multiple values, we have prevented this from happening by selecting the largest value when more than one is selected. Also, it was requested by the shifters for us to remove the red line from the sbit graphs.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
This bug prevented tests from being run at qc7 and needed to be addressed.

## How Has This Been Tested?
```bash
anaSBitThresh.py /data/bigdisk/GEM-Data-Taking/GE11_QC8///sbitRate/channelOR//2020.02.12.11.33/SBitRateData.root

echo "File 1"
echo "This tests how the code handles a vfat with a bad vfat"
echo "Please make sure (2,5,4) is filled in your system_specfic_constants.py file"

# This file causes issues because of a zero slope issue
anaSBitThresh.py /data/bigdisk/GEM-Data-Taking/GE11_QC8//sbitRate/channelOR/2019.10.25.13.59/SBitRateData.root

echo "File 2"
echo "This tests how the code handles a detector with missing vfats"
echo "Please make sure (1,2,0) is filled in your system_specfic_constants.py file"

# This file is from the coffin where the detector is missing vfats
anaSBitThresh.py /data/bigdisk/GEM-Data-Taking/GE11_Integration//sbitRate/channelOR//2019.10.24.17.45/SBitRateData.root
```

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bregnery** [comment]
This is to address issue [#288]

**Comment by cgalloni** [comment]
@bregnery, can you solve the conflicting files? Maybe with a rebase?

**Comment by bregnery** [comment]
Ah, yes, sorry, I rebased to a previous tag

File: utils/anautilities.py [hunk 1]
```diff
@@ -556,7 +556,8 @@ def findInflectionPts(graph):
     # We are defining the inflection point as the point where the most negative gradient sum begins
     try:
         inflxGrad = negGrad[bigIdx][0]
-        inflxIdx = np.where(grad == inflxGrad) #return the index at the specified value
+        inflxIdices = np.where(grad == inflxGrad) #return the index at the specified value
+        inflxIdx = np.amax(inflxIdices) # if there are multiple possible points, take the largest one
```
Comment (lpetre-ulb) [inline utils/anautilities.py]: Is the variable `inflxIdices` necessary?

File: utils/threshAlgos.py [hunk 2]
```diff
@@ -1452,7 +1452,7 @@ def getDetName(entry):
                 # channelor case
                 if perchannel == False :
 	            dict_dacInflectPts[dacName][ohKey][vfat] = findInflectionPts(dict_Rate1DVsDACNameX[dacName][ohKey][vfat])
-                    inflectTable.append([ohKey, vfat, dict_dacInflectPts[dacName][ohKey][vfat][0][0] ])
+                    inflectTable.append([ohKey, vfat, dict_dacInflectPts[dacName][ohKey][vfat][0] ])
```
Comment (lpetre-ulb) [inline utils/threshAlgos.py:1457]: I guess this fixes the same bug as the one described in https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/268?

**Review by lpetre-ulb — DISMISSED** [review]
Non blocking comments.

File: utils/anautilities.py (hunk 1 above)
Comment (bregnery) [reply utils/anautilities.py]: Good point!

File: utils/threshAlgos.py [hunk 3]
```diff
@@ -1498,24 +1498,13 @@ def getDetName(entry):
                 for vfat in range(0,nVFATS):
                     canv_Summary1D.cd(chamber_vfatPos2PadIdx[gemType][vfat]).SetLogy()
 
-                    # make sure the inflection point is there
```
Comment (jsturdy) [inline utils/threshAlgos.py:1501]: I would probably keep the line drawing functionality, but disable it by default if someone looking at the plot finds it too distracting

File: utils/threshAlgos.py (hunk 2 above)
Comment (bregnery) [reply utils/threshAlgos.py:1457]: Yeah, now that it is restricted to have one index, the data structure changed a bit.

File: utils/threshAlgos.py (hunk 3 above)
Comment (bregnery) [reply utils/threshAlgos.py:1501]: Okay, I have not added input options in python, so I will have to take a look at how to do this.

Comment (bregnery) [reply utils/threshAlgos.py:1501]: Should I also disable by default the printing of the table? (The table will still be printed to the file)

**Comment by bregnery** [comment]
I have added an option inside anautilities for drawing the lines on the graph and for printing the table output in the terminal. These options default to false. The inflection point values are still stored in a `.txt` file by default.

I have tested this on sbit scans that have multivalued inflection points, missing vfats, and bad sbit lines.

File: utils/threshAlgos.py (hunk 3 above)
Comment (jsturdy) [reply utils/threshAlgos.py:1501]: were there complaints about the table?
i think there are other tools that are printing tables to the output, so i wouldn't change that

Comment (bregnery) [reply utils/threshAlgos.py:1501]: Okay, then I will default that option to true and leave it as an option if people complain about the table in the future :)

**Review by cgalloni — APPROVED** [review]

**Review by jsturdy — APPROVED** [review]

**Pull request merged**